packages/webbundle-webpack-plugin/src/index.ts

```diff
@@ -95,17 +95,32 @@ export class WebBundlePlugin implements WebpackPluginInstance {
 
   apply = (compiler: Compiler) => {
     if (isWebpackMajorV4()) {
-      compiler.hooks.emit.tap(this.constructor.name, this.process);
+      compiler.hooks.emit.tapAsync(
+        this.constructor.name,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        async (compilation: Compilation, callback: any) => {
+          await this.process(compilation);
+
+          // Invokes webpack provided callback after functionality is complete.
+          callback();
+        }
+      );
     } else {
       compiler.hooks.thisCompilation.tap(
         this.constructor.name,
         (compilation: Compilation) => {
-          compilation.hooks.processAssets.tap(
+          compilation.hooks.processAssets.tapAsync(
```

Comment:
I wonder if `tapPromise` would be supported in all of our supported Webpack v5 versions?